### ISSUE TITLE
Absence of mailhog should not slow down scripts

### ISSUE THREAD

**Issue by dealancer** [docksal/docksal#495]
**Description**

There is a problem sending mail which causes a timeout and delays of the scripts where mail is sent.

**Steps to reproduce the issue:**

1.  Updated cli image to `2.0-php7.2` or ` edge-php7.2` and web image to `2.0-apache2.4` or `edge-apache2.4`.
2. Run `fin init`
3.  Run any cli command that is sending mail, e.g. `wp core install`.

**Describe the results you received:**

```
error sending mail
2018/04/05 17:59:42 dial tcp 141.161.99.99:1025: getsockopt: connection timed out
```

**Describe the results you expected:**

There should be no time outs when sending emails. I do not care much about email delivery, but it should not be slowing down scripts.

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
---------------------
COMPOSE_PROJECT_NAME_SAFE: guwordpress
COMPOSE_FILE:
/Users/vm386/.docksal/stacks/volumes-bind.yml
/Users/vm386/.docksal/stacks/stack-default.yml
/Users/vm386/Projects/docksal/guwordpress/.docksal/docksal.yml
ENV_FILE:
/Users/vm386/Projects/docksal/guwordpress/.docksal/docksal.env

PROJECT_ROOT: /Users/vm386/Projects/docksal/guwordpress
DOCROOT: docroot
VIRTUAL_HOST: guwordpress.docksal
VIRTUAL_HOST_ALIASES: *.guwordpress.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32773

Docker Compose configuration
---------------------
services:
  browser:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: browser
    image: selenium/standalone-chrome
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      DOCROOT: docroot
      HOST_GID: '20'
      HOST_UID: '502'
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: docksal/cli:edge-php7.2
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - project_root:/var/www:rw,nocopy
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_DATABASE: default
      MYSQL_PASSWORD: user
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.1-mysql-5.6
    ports:
    - 03306/tcp
    volumes:
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:edge-apache2.4
    labels:
      io.docksal.project-root: /Users/vm386/Projects/docksal/guwordpress
      io.docksal.virtual-host: guwordpress.docksal,*.guwordpress.docksal
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  docksal_ssh_agent:
    external: true
  project_root:
    driver: local
    driver_opts:
      device: /Users/vm386/Projects/docksal/guwordpress
      o: bind
      type: none

---------------------
```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
fin sysinfo output
███  OS & BASICS
Darwin UIS-AL-LNFD58.local 16.7.0 Darwin Kernel Version 16.7.0: Tue Jan 30 11:27:06 PST 2018; root:xnu-3789.73.11~1/RELEASE_X86_64 x86_64
Mac OS X 10.12.6
fin version: 1.27.0
Mode:  VirtualBox VM
███  INSTANCES
NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
docksal   *        virtualbox   Running   tcp://192.168.64.100:2376           v17.06.0-ce   
███  DOCKER
DOCKER_HOST:	tcp://192.168.64.100:2376

Docker: Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:31:53 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:51:55 2017
 OS/Arch:      linux/amd64
 Experimental: false
███  DOCKER COMPOSE
Docker Compose: docker-compose version 1.14.0, build c7bdf9e
docker-py version: 2.3.0
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016
███  DOCKER MACHINE
docker-machine version 0.12.2, build 9371605
███  DOCKER: IMAGES
REPOSITORY                   TAG                 IMAGE ID            CREATED             SIZE
docksal/cli                  edge-php7.2         5413f1758a7a        5 days ago          906MB
docksal/cli                  2.0-php7.2          8f42648197b3        2 weeks ago         842MB
docksal/cli                  php7.2              2a358a42f82a        2 weeks ago         842MB
selenium/standalone-chrome   latest              825e97a855ef        2 months ago        825MB
guwordpress_cli              latest              c043f04e1d7e        3 months ago        818MB
docksal/cli                  1.3-php7            19bdddfeab89        6 months ago        759MB
docksal/web                  2.1-apache2.4       32c1d32eba6d        7 months ago        87.5MB
docksal/web                  edge-apache2.4      1b939e80c6f3        7 months ago        87.5MB
docksal/web                  2.0-apache2.4       7f75db9a6aa7        8 months ago        87.5MB
docksal/vhost-proxy          1.1                 898fb21e7f2d        8 months ago        158MB
docksal/db                   1.1-mysql-5.7       8d323c436c28        8 months ago        412MB
docksal/db                   1.1-mysql-5.6       6feb79af9597        8 months ago        298MB
mailhog/mailhog              latest              669347153085        11 months ago       19.1MB
docksal/dns                  1.0                 20c7a535479d        14 months ago       61.6MB
docksal/ssh-agent            1.0                 148220cc033d        14 months ago       12.2MB
docksal/mysql                5.7                 8bebc7186864        18 months ago       369MB
███  DOCKER: CONTAINERS
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS                    PORTS                                                    NAMES
3f964952f19b        docksal/web:edge-apache2.4   "httpd-foreground"       14 minutes ago      Up 14 minutes             80/tcp, 443/tcp                                          guwordpress_web_1
b34b5e6c2259        docksal/db:1.1-mysql-5.6     "/entrypoint.sh my..."   14 minutes ago      Up 14 minutes             0.0.0.0:32773->3306/tcp                                  guwordpress_db_1
f9fd844fda09        docksal/cli:edge-php7.2      "/opt/startup.sh s..."   14 minutes ago      Up 14 minutes (healthy)   22/tcp, 3000/tcp, 9000/tcp                               guwordpress_cli_1
9a17274389b7        selenium/standalone-chrome   "/opt/bin/entry_po..."   14 minutes ago      Up 14 minutes             4444/tcp                                                 guwordpress_browser_1
d1f49092effc        docksal/ssh-agent:1.0        "/run.sh ssh-agent"      3 months ago        Up 2 hours                                                                         docksal-ssh-agent
9acda045e292        docksal/dns:1.0              "/opt/entrypoint.s..."   3 months ago        Up 2 hours                192.168.64.100:53->53/udp                                docksal-dns
38c1466464e5        docksal/vhost-proxy:1.1      "docker-entrypoint..."   3 months ago        Up 2 hours                192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
███  VIRTUALBOX
5.1.22r115126
███  NETWORK INTERFACES
Name:            vboxnet0
GUID:            786f6276-656e-4074-8000-0a0027000000
DHCP:            Disabled
IPAddress:       192.168.64.1
NetworkMask:     255.255.255.0
IPV6Address:     
IPV6NetworkMaskPrefixLength: 0
HardwareAddress: 0a:00:27:00:00:00
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-vboxnet0

███  NFS EXPORTS

# <ds-nfs docksal
/Users 127.0.0.1 192.168.64.100 -alldirs -maproot=0:0
# ds-nfs>
███  MOUNTS
tmpfs on / type tmpfs (rw,relatime,size=1845260k)
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,relatime)
devpts on /dev/pts type devpts (rw,relatime,mode=600,ptmxmode=000)
tmpfs on /dev/shm type tmpfs (rw,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
/dev/sda1 on /mnt/sda1 type ext4 (rw,relatime,data=ordered)
cgroup on /sys/fs/cgroup type tmpfs (rw,relatime,mode=755)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,relatime,cpuset)
cgroup on /sys/fs/cgroup/cpu type cgroup (rw,relatime,cpu)
cgroup on /sys/fs/cgroup/cpuacct type cgroup (rw,relatime,cpuacct)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,relatime,blkio)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,relatime,memory)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,relatime,devices)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls type cgroup (rw,relatime,net_cls)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,relatime,perf_event)
cgroup on /sys/fs/cgroup/net_prio type cgroup (rw,relatime,net_prio)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,relatime,hugetlb)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,relatime,pids)
/dev/sda1 on /mnt/sda1/var/lib/docker/aufs type ext4 (rw,relatime,data=ordered)
shm on /mnt/sda1/var/lib/docker/containers/d1f49092effce03abd75bd7e5cebe7460b178a02ef5df72a0aed484282d0b42f/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
none on /mnt/sda1/var/lib/docker/aufs/mnt/4df767afceac34de4195dd7e51836ba5c6e56dbb85f38653899c9c2af19e4d50 type aufs (rw,relatime,si=dbb1ac82ca856693,dio,dirperm1)
none on /mnt/sda1/var/lib/docker/aufs/mnt/dd5b090522ec31104e0d5934a490e66ca3dde6c12912473f29db683e3a03c87e type aufs (rw,relatime,si=dbb1ac82c59bf693,dio,dirperm1)
nsfs on /var/run/docker/netns/8b1e1014d547 type nsfs (rw)
shm on /mnt/sda1/var/lib/docker/containers/38c1466464e54966fb4e51c58961ff5e6d09162f16e41613140adde5df84a053/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
shm on /mnt/sda1/var/lib/docker/containers/9acda045e292d3269308ef47a278e30b0675d39bd4fc830362ac9957098e755b/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
nsfs on /var/run/docker/netns/2a2d1d4bfdd3 type nsfs (rw)
nsfs on /var/run/docker/netns/07f493f9f59d type nsfs (rw)
192.168.64.1:/Users on /Users type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)
none on /mnt/sda1/var/lib/docker/aufs/mnt/77e07f3a89ec3278e9624c4cfe2f843654b17ad07191a44337953857eda32eed type aufs (rw,relatime,si=dbb1ac828504d693,dio,dirperm1)
shm on /mnt/sda1/var/lib/docker/containers/9a17274389b7b24827d3a904193ffc36f9618977d07e9f6fbd72cc892f928475/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
none on /mnt/sda1/var/lib/docker/aufs/mnt/ec40377b7ad0d0d0227460c462dd94c11774b3de9ef5b490c5be5778a1d18ae6 type aufs (rw,relatime,si=dbb1ac82bd6f4693,dio,dirperm1)
nsfs on /var/run/docker/netns/4b8da40098e6 type nsfs (rw)
none on /mnt/sda1/var/lib/docker/aufs/mnt/db598dc8ef6ce0b1712adf5b1d87417a58e97ae922e000302eb65168a1d46537 type aufs (rw,relatime,si=dbb1ac82bd6f9693,dio,dirperm1)
shm on /mnt/sda1/var/lib/docker/containers/f9fd844fda099c48fffbaabbb2f27d331548d70a6da5ad5afca0b224644ce70e/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
192.168.64.1:/Users/vm386/Projects/docksal/guwordpress on /mnt/sda1/var/lib/docker/volumes/guwordpress_project_root/_data type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)
nsfs on /var/run/docker/netns/931238e77de9 type nsfs (rw)
shm on /mnt/sda1/var/lib/docker/containers/b34b5e6c22594d58e2b03bb90b09da714402f03f4fc4867d0d34e09cb7e49b52/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
nsfs on /var/run/docker/netns/fea9c9a541e2 type nsfs (rw)
none on /mnt/sda1/var/lib/docker/aufs/mnt/4e1ac54979b9e7dd16a1f060dd3a9b426243af78e144edfb1a21be01a93021c1 type aufs (rw,relatime,si=dbb1ac82b5ed3693,dio,dirperm1)
shm on /mnt/sda1/var/lib/docker/containers/3f964952f19b7f5deb70d21711613d7b842e7d1e167b78278d1bccfdb287687f/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=65536k)
nsfs on /var/run/docker/netns/3bf99f8ba6b9 type nsfs (rw)
 ```

</details>

**Comment by lmakarov** [comment]
Starting with `docksal/cli` v2.0.0 we configure mailhog by default in PHP settings
https://github.com/docksal/service-cli/issues/22

```
sendmail_path = '/usr/local/bin/mhsendmail --smtp-addr=mail:1025'
```

This used to be a dummy command to suppress errors:

```
sendmail_path = '/bin/true'
```

Then you would have to override it to point to the mailhog container (settings above) when adding the [mailhog](https://docs.docksal.io/en/master/tools/mailhog/) integration to a project.

The error is safe to ignore.

To suppress it, you can override PHP settings at the project level via `.docksal/etc/php/php.ini` like this

```
; Mail settings
sendmail_path = '/bin/true'
```

Here's how it can be suppressed ad-hoc when doing `drush site-install`
https://github.com/docksal/drupal8/commit/9eca2bc54e052a648e8667ed53b0f595a30d63ba

**Comment by lmakarov** [comment]
Now that I'm giving it another thought, we could just use the following as a universal configuration (with or without mailhog container available):

```
sendmail_path = 'bash -c "/usr/local/bin/mhsendmail --smtp-addr=mail:1025 || true"'
```

or just

```
sendmail_path = '/usr/local/bin/mhsendmail --smtp-addr=mail:1025 || true'
```

**Comment by dealancer** [comment]
@lmakarov thanks for quick reply.

This one is a good workaround:

`sendmail_path = '/bin/true'`

Following one produces huge timeout (several minutes) with no error message:

`sendmail_path = '/usr/local/bin/mhsendmail --smtp-addr=mail:1025 || true'`

**Comment by achekulaev** [comment]
@dealancer 

> Following one produces huge timeout (several minutes) with no error message:

Please provide full list of steps to reproduce. For instance it is not clean what project you are setting up, whether it is Docksal Wordpress example project or something else.

**Comment by achekulaev** [comment]
@lmakarov suppressing the error is not always good too (it would suppress any errors, even valid mailhog errors like invalid e-mail)

I'm afraid we would have to use some variable like `SENDMAIL_ENABLED` to control the behavior.

**Comment by dealancer** [comment]
@achekulaev this is a custom WordPress project, but I think if you use `Docksal Wordpress` you just run `fin init`, also you may need to update `docksal.env` to use images I mentioned.

**Comment by achekulaev** [comment]
@dealancer I checked out https://github.com/docksal/wordpress, changed image to `docksal/cli:2.0-php7.2`, ran `fin init` and did not encounter any timeout on that step.

```
 Step 3  Installing site...
error sending mail
2018/04/05 20:06:47 dial tcp: lookup mail on 127.0.0.11:53: no such host
```

This happened instantly.

**Comment by lmakarov** [comment]
@dealancer I thing the timeout is related to you LAN setup. Specifically, in your environment `mail` resolves to `141.161.99.99`, which may lead to `mhsendmail` trying to communicate to that host, then failing.

I believe that in your case both variants (the default and with the `|| true` override) will have the same result - the long timeout you are experiencing.

**Comment by achekulaev** [comment]
@dealancer was the issue resolved?
